### PR TITLE
gui: Implement TransactionRecord::Message

### PR DESCRIPTION
**Pull request by jamescowens**
This allows the display of a message icon for sent or received GRC with a message attached.

### CODE REVIEW

File: src/qt/transactionrecord.cpp [hunk 1]
```diff
@@ -58,8 +58,19 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet *
...
+    bool fContractPresent = false;
+    NN::ContractType ContractType;
+
+    if (!wtx.GetContracts().empty())
+    {
+         const auto& contract = wtx.GetContracts().begin();
+         fContractPresent = true;
+         ContractType = contract->m_type.Value();
+    }
+
```
Comment (cyrossignol) [inline src/qt/transactionrecord.cpp:76]: Any reason to pull this out still? It looks like it's only being used in the original.

Comment (cyrossignol) [reply src/qt/transactionrecord.cpp:76]: Oh wait, I see it used below there.

**Review by cyrossignol — APPROVED** [review]

File: src/qt/transactionrecord.cpp (hunk 1 above)
Comment (jamescowens) [reply src/qt/transactionrecord.cpp:76]: Yep. Used twice...

Comment (jamescowens) [reply src/qt/transactionrecord.cpp:76]: And it is better to do it once per transaction rather than once every output...

**Pull request merged**